homeassistant/components/bmw_connected_drive/sensor.py

```diff
@@ -228,6 +574,7 @@ def update(self) -> None:
         _LOGGER.debug("Updating %s", self._vehicle.name)
         vehicle_state = self._vehicle.state
         vehicle_last_trip = self._vehicle.state.last_trip
+        vehicle_all_trips = self._vehicle.state.all_trips
```

Comment:
It's only used if `self._service == SERVICE_ALL_TRIPS` so no need to fetch earlier.